aspnetcore-webapi/RemoteHostIpAddress/RemoteHostIpAddressTests/RemoteHostIpAddressTests.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using RemoteHostIpAddress.Controllers;
+using System.Net;
+using Xunit;
+
+namespace RemoteHostIpAddressTests
+{
+    public class RemoteHostIpAddressTests
+    {
+        private readonly WeatherForecastController _controller;
+
+        public RemoteHostIpAddressTests()
+        {
+            _controller = new WeatherForecastController(new NullLogger<WeatherForecastController>());
+        }
+
+        [Fact]
+        public void WhenGetRemoteHostIpAddressUsingRemoteIpAddress_ThenReturnsValidIpAddress()
+        {
+            // Arrange
+            var connection = new Mock<ConnectionInfo>();
+            connection.SetupGet(c => c.RemoteIpAddress).Returns(IPAddress.Parse("192.168.1.1"));
+
+            var httpContext = new Mock<HttpContext>();
+            httpContext.SetupGet(h => h.Connection).Returns(connection.Object);
+
+            _controller.ControllerContext.HttpContext = httpContext.Object;
+
+            // Act
+            var result = _controller.GetRemoteHostIpAddressUsingRemoteIpAddress();
+
+            // Assert
+            Assert.Equal(IPAddress.Parse("192.168.1.1"), result);
+        }
+
+        [Fact]
+        public void WhenGetRemoteHostIpAddressUsingXForwardedFor_ThenReturnsValidIpAddress()
+        {
+            // Arrange
+            var headerDictionary = new HeaderDictionary
+            {
+                { "X-Forwarded-For", "192.168.1.1" }
+            };
+
+            var httpContext = new Mock<HttpContext>();
+            httpContext.SetupGet(h => h.Request.Headers).Returns(headerDictionary);
+
+            _controller.ControllerContext.HttpContext = httpContext.Object;
+
+            // Act
+            var result = _controller.GetRemoteHostIpAddressUsingXForwardedFor();
+
+            // Assert
+            Assert.Equal(IPAddress.Parse("192.168.1.1"), result);
+        }
+
+        [Fact]
+        public void WhenGetRemoteHostIpAddressUsingXForwardedFor_WithInvalidIp_ThenReturnsNull()
```

Comment:
The name of the method doesn't conform to our guidelines, the underscore should only be used to separate between Given, When and Then. Here it's used to separate the With. This method should be named:  
`WhenGetRemoteHostIpAddressUsingXForwardedForWithInvalidIp_ThenReturnsNull` instead.

Reply:
sure, thanks for the reminder :)